### ISSUE TITLE
Post Types Custom Capabilities

### ISSUE THREAD

**Issue by papazetis** [AdvancedCustomFields/acf#805]
I have created a post type with custom capabilities but when I create a post I get the following error:
"Sorry, you are not allowed to edit this post."

![image](https://user-images.githubusercontent.com/3592665/229811359-1447beba-4033-485a-a8ff-affff5266aa8.png)

User role editor
![image](https://user-images.githubusercontent.com/3592665/229811600-bfde3421-0aca-4dbe-8e52-e2b76409ac99.png)

These are the custom capabilities when I create the post type by `register_post_type`
![image](https://user-images.githubusercontent.com/3592665/229812491-d766a3e4-e429-4ba8-9d5d-28e5cc0017fb.png)

**Comment by papazetis** [comment]
Looks like the `map_meta_cap` is false which is causing this issue

**Comment by mattgrshaw** [comment]
Hi @papazetis thanks for the report.

ACF doesn't currently have a UI setting for `map_meta_cap` or pass this in by default (this may change in a future release), but you can filter this in the meantime with something like the below:

```
function add_map_meta_cap( $args, $post_type ) {
	if ( ! isset( $args['map_meta_cap'] ) ) {
		$args['map_meta_cap'] = true;
	}

	return $args;
}
add_filter( 'acf/post_type/registration_args', 'add_map_meta_cap', 10, 2 );
```

**Comment by papazetis** [comment]
Thank you @mattgrshaw for the update and the solution.
I ll do that for now but, correct me if I am wrong, without `map_meta_cap` setting on the UI, the custom capabilities are useless

**Comment by mattgrshaw** [comment]
@papazetis We just released ACF/ACF PRO 6.1.3 which will set `map_meta_cap` to true if "Rename Capabilities" is enabled with custom capabilities passed in.

We'll be keeping an eye on this and may still consider a UI setting in the future if we feel it's necessary.
